cmd/antrea-controller/app/config/config.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package controller
+package config
```

Comment:
Why shall we need to introduce this new directory?

Reply:
yes, that was intentional, so that it can be imported easily from e2e test cases
I see no reason to move it back?

Reply:
I saw `kubernetes` and some other projects using this similar code architecture in the cmd package:
`https://github.com/kubernetes/kubernetes/tree/master/cmd/kube-controller-manager/app`.
I think it is better to put the configs in the entry but the pkg packages.

Reply:
Regardless of whether Antrea could really benefit from following the same cmd structure as K8s's, it does't make sense to do such refactoring in a PR for unit test. If you do see necessities of making that change, you should open a separate PR and describe why the change is necessary, "kubernetes and some other projects do it" is a good point if we are starting from scratch, but is not so good at the current stage after Antrea has used the current structure for 3+ years without meeting real issue.

Reply:
Yes, it works well with the existing structure. Maybe we can use a better structure as there are ~300 lines in the controller entry function and ~800 lines in the agent entry function. And it will be longer and longer when more and more features will be added. I think it’s a good chance to refactor these while adding test cases.

Reply:
@wenqiq could you help to address all previous comments? Let's include this in release-1.12.